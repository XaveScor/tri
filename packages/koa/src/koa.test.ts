import Koa from 'koa';
import Request from 'supertest';
import { createDrzewoMiddleware } from './index';
import { ChildWidget } from './fixtures/ChildWidget';

describe('koa', () => {
  it('base middleware', async () => {
    const app = new Koa();

    app.use(createDrzewoMiddleware(ChildWidget, { a: '123' }));

    const response = await Request(app.callback()).get('/');

    expect(response.body).toMatchSnapshot();
  });
});
