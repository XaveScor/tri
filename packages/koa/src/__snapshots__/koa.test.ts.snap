// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`koa base middleware 1`] = `
Object {
  "data": Array [
    49,
    50,
    51,
  ],
  "type": "Buffer",
}
`;
